tests/test_selenium.py

```diff
@@ -38,37 +40,73 @@ def driver(request):
 
 
 def test_notebook(driver):
+    """Ensures a notebook can be created and connected to."""
+
     driver, wait, url = driver
 
     notebook_name = 'ci-test-' + ''.join(choices(ascii_lowercase, k=10))
 
     # Click "New Server" button
-    driver.find_element_by_id("add-nb").click()
-    wait.until(url_to_be(url + 'new'))
+    driver.find_element_by_id("newResource").click()
+    wait.until(EC.url_to_be(url + 'new'))
 
     # Enter server name
-    name_input = driver.find_element_by_id('mat-input-0')
+    name_input = driver.find_element_by_css_selector('input[placeholder="Name"]')
     name_input.send_keys(notebook_name)
     name_input.click()
 
-    driver.find_element_by_css_selector('button[type="submit"]').click()
-    wait.until(url_to_be(url))
-
-    # Ensure that the Notebook resource is created. It won't finish setting up
-    # due to a missing `default-editor` ServiceAccount that it relies on that's
-    # created by the kubeflow-profiles service.
-    row = wait.until(
-        lambda d: next(
-            row for row in d.find_elements_by_css_selector('.mat-row') if notebook_name in row.text
+    # Click submit on the form. Sleep for 1 second before clicking the submit
+    # button because shiny animations that ignore click events are simply a must.
+    # Note that that was sarcasm. If you're reading this, please don't shit up
+    # the web with braindead technologies.
+    submit = wait.until(
+        EC.presence_of_element_located(
+            (By.CSS_SELECTOR, ".form--button-margin.mat-primary:not(disabled)")
         )
     )
-    # Delete notebook
-    row.find_element_by_css_selector('button:last-child').click()
-    driver.find_element_by_css_selector('.yes').click()
-    row = wait.until_not(
-        lambda d: next(
-            row
-            for row in d.find_elements_by_css_selector('.mat-row')
-            if f'check_circle\n{notebook_name}' in row.text
-        )
+    sleep(1)
+    submit.click()
+    wait.until(EC.url_to_be(url))
+
+    # Since upstream doesn't use proper class names or IDs or anything, find the
+    # <tr> containing elements that contain the notebook name and `ready`, signifying
+    # that the notebook is finished booting up. Returns a reference to the containing
+    # <tr> element. The result is a fairly unreadable XPath reference, but it works 🤷
```

Comment:
Not sure if we want to start having emojis in our codebase

Reply:
We already do here: https://github.com/canonical/bundle-kubeflow/blob/master/tests/test_selenium.py#L22

Incidentally also added to placate my annoyance with having to deal with broken web technologies